doc/rst/developer/bestPractices/EditionsDev.rst

```diff
@@ -18,8 +18,117 @@ When creating new editions, be sure to:
+We consider a feature to be stable if it is not marked unstable (either directly
+or indirectly by being included in a module that is marked unstable).
+
+* **Breaking Changes**
+
+   * altering the behavior of a stable feature
+
+   * removal of a stable feature
+
+* **Non-Breaking Changes**
+
+   * altering the behavior of an unstable feature
+
+   * deprecation or removal of an unstable feature
+
+   * addition of a new feature
+
+.. note::
+
+   On a case by case basis, some changes that would normally be considered
+   non-breaking may be treated as breaking changes. For example, if an
+   unstable feature is widely used in practice, changing its behavior may be
+   considered a breaking change.
+
+The following provides several scenarios of language evolution. In
+this, we do not distinguish between language features and standard library
+features.
+
+* **Changing the behavior of a stable feature** (`Breaking Change`)
+
+    A stable feature should only have its behavior change in the ``preview``
+    edition. The ``preview`` edition collects changes that will be included in
+    the next edition.
+
+    .. note::
+
+        An unimplemented feature is the flag described in
+        `#27336 <TransitionFlag>`_, which will warn users
+        in a given edition about code that will need to be migrated in the next
+        edition.
+
+* **Removal of a stable feature** (`Breaking Change`)
+
+    A stable feature should only be removed in the ``preview`` edition. The
+    ``preview`` edition collects changes that will be included in the next
+    edition. The act of marking a stable feature as removed in a future edition
+    enables deprecation warnings for the feature in the current edition
+    (when requested).
+
+    .. note::
+
+        An unimplemented feature is the flag described in
+        `#27336 <TransitionFlag>`_, which will warn users
+        in a given edition about code that will need to be migrated in the next
+        edition.
+
+* **Changing the behavior of an unstable feature** (`Non-Breaking Change`)
+
+    Unstable features are not considered part of an edition, and so their
+    behavior can be changed at any time without being associated with an
+    edition. However, the first Chapel compiler release to change the behavior
+    of an unstable feature should warn users that the feature has changed in
+    some way. Later Chapel compiler releases may remove the warning.
+
+* **Deprecation (and subsequent removal) of an unstable feature** (`Non-Breaking Change`)
+
+    Unstable features are not considered part of an edition, and so they can be
+    deprecated at any time without being associated with an edition. However,
+    the first Chapel compiler release to deprecate or remove an unstable
+    feature should warn users that the feature has been deprecated or removed.
+    Later Chapel compiler releases may remove the warning and the feature.
+    Unstable features should not be removed without warning.
```

Comment:
Should this be rephrased to refer and link to our normal deprecation lifecycle? 
https://chapel-lang.org/docs/developer/bestPractices/Deprecation.html

Currently 

> the first Chapel compiler release to deprecate or remove an unstable feature

Makes it a little ambiguous as to whether you're allowed to just remove an unstable feature without deprecating it first.

Reply:
I think the deprecation policy may also need to be tweaked, I'll take a look and make sure to link back to it